### PR TITLE
feat: implement navigation bar and tabs for main account screens

### PR DESCRIPTION
**Pull request by simonheys**
This PR partially implements the new navigation bar and tab style in the main Account screens. These screens use a new root-level container which simplifies responsive and local scrolling behaviour and paves the way for future features.

https://user-images.githubusercontent.com/175607/199216936-18008cae-2e07-4457-bdd4-99f10203ce1a.mov

### CODE REVIEW

File: packages/extension/e2e/setups/accountWithTestnetEth.ts [hunk 1]
```diff
@@ -25,8 +25,8 @@ export async function setupNewAccountWithTestnetEth(
   // page is now closed
   await openExtension(page, context)
   await continueNewWalletAfterOnboarding(page)
-  await navigateFromAccountToAccountList(page)
   await switchNetwork(page, "Localhost")
+  // await navigateFromAccountToAccountList(page)
```
Comment (simonheys) [inline packages/extension/e2e/setups/accountWithTestnetEth.ts:29]:
```suggestion
```

File: packages/extension/src/ui/components/ContentContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { AbsoluteBox, scrollbarStyle } from "@argent/ui"
+import { ComponentProps, forwardRef } from "react"
+
+import { NavigationBarHeight } from "./NavigationBar"
+import { TabBarHeight } from "./TabBar"
+
+interface ContentContainerProps extends ComponentProps<typeof AbsoluteBox> {
+  /** top should be inset for NavigationBar component */
+  navigationBarInset?: boolean
+  /** top should be inset for TabBar component */
```
Comment (simonheys) [inline packages/extension/src/ui/components/ContentContainer.tsx:10]:
```suggestion
  /** bottom should be inset for TabBar component */
```

File: packages/extension/src/ui/components/NavigationBar.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import { AbsoluteFlex, H6, IScroll, icons } from "@argent/ui"
```
Comment (janek26) [inline packages/extension/src/ui/components/NavigationBar.tsx:1]: what's the reaso to keep these components in the extension package and not generalize them into the ui package?

File: packages/ui/src/hooks/useScroll.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  scrollTop: number
+  scrollLeft: number
+}
+
+/**
+ * Hook to monitor and return the scroll position of a div.
+ *
+ * @example
+ * ```ts
+ * // pass the scroll position of `ContentContainer` into `NavigationBar`
```
Comment (janek26) [inline packages/ui/src/hooks/useScroll.ts:13]: why is this needed, couldnt that be replaced with position absolute/sticky?

File: packages/extension/src/ui/components/NavigationBar.tsx (hunk 3 above)
Comment (janek26) [reply packages/extension/src/ui/components/NavigationBar.tsx:1]: I mean the rest of the file, sorry

File: packages/ui/src/hooks/useScroll.ts (hunk 4 above)
Comment (simonheys) [reply packages/ui/src/hooks/useScroll.ts:13]: The `NavigationBar` component in the new design changes its background from transparent to solid based on whether another component is scrolled or not (you can just about see this in the video posted above). This hook provides a simple way for those components to share and act on scroll offset.

File: packages/extension/src/ui/components/NavigationBar.tsx (hunk 3 above)
Comment (simonheys) [reply packages/extension/src/ui/components/NavigationBar.tsx]: Reworked these and moved into `@argent/ui`

**Review by janek26 — APPROVED** [review]
